db/commitlog/commitlog.cc

```diff
@@ -1586,6 +1586,8 @@ future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager:
 }
 
 future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager::new_segment() {
+    gate::holder g(_gate);
+
```

Comment:
In `shutdown()` we both set `_shutdown` and close `_gate` so if the gate is closed here we won't get to the statement below.
Why not hold the gate after checking `_shutdown`?

Reply:
Just same order as old code. Either is fine I guess...

Reply:
I'd rather not change existing errors as tests may depend on them.

Reply:
Then this is correct, as it is the order we took things before (gate was held in calling method).

Reply:
Ah, I see